packages/hint-button-type/src/hint.ts

```diff
@@ -23,6 +23,20 @@ export default class ButtonTypeHint implements IHint {
 
     public constructor(context: HintContext) {
 
+        let inAForm = false;
+
+        const traverseDown = (evt: TraverseDown) => {
+            if (evt.element.nodeName === 'form') {
+                inAForm = true;
+            }
+        };
+
+        const traverseUp = (evt: TraverseUp) => {
+            if (evt.element.nodeName === 'form') {
```

Comment:
This seems a bit indirect. 

How about walking the `parentElement` chain on the `HTMLElement` returned by the `ElementFound` event instead?

Reply:
Not a big fun of that tbh.
Is it just because this is the first time we use this event and you wanted to get rid of them? 😋

Reply:
> Not a big fun of that tbh.

Why?

> Is it just because this is the first time we use this event and you wanted to get rid of them?

No 😋 But seriously, I'm concerned about storing contextual state in a "global" context and taking a dependency on the order in which events will fire.

For example, as-is this code will break if encountering a nested form. Not saying a nested form is a good idea, but I'm sure we'll hit it on some site somewhere...

Reply:
Hope you like this version better 😋